apps/test/unit/code-studio/components/progress/LessonLockDialogTest.jsx

```diff
@@ -1,112 +1,188 @@
 import React from 'react';
-import {shallow} from 'enzyme';
+import {mount} from 'enzyme';
 import {expect} from '../../../../util/reconfiguredChai';
 import sinon from 'sinon';
 import {UnconnectedLessonLockDialog as LessonLockDialog} from '@cdo/apps/code-studio/components/progress/LessonLockDialog';
 import {LockStatus} from '@cdo/apps/code-studio/lessonLockRedux';
 
+// This * import allows us to stub out the SectionSelector component
+import * as SectionSelector from '@cdo/apps/code-studio/components/progress/SectionSelector';
+
+const fakeSectionId = 42;
+
 const MINIMUM_PROPS = {
-  isOpen: false,
+  isOpen: true,
   handleClose: () => {},
   initialLockStatus: [],
-  selectedSectionId: null,
+  selectedSectionId: fakeSectionId,
   saving: false,
   saveDialog: () => {}
 };
 
-const fakeSectionId = 1;
+// Helper function to get the list of rows in the student table.
+const getStudentRows = wrapper =>
+  wrapper.find('#ui-test-student-table tbody tr');
+
+// Helper function to verify that the radio buttons in the given student row
+// correctly reflect the expected lock status.
+const verifyRowLockStatus = (row, expectedLockStatus) => {
+  const radioButtons = row.find('input');
+  const lockedSelected = radioButtons.at(0).props().checked;
+  expect(lockedSelected).to.equal(expectedLockStatus === LockStatus.Locked);
+  const editableSelected = radioButtons.at(1).props().checked;
+  expect(editableSelected).to.equal(expectedLockStatus === LockStatus.Editable);
+  const readonlySelected = radioButtons.at(2).props().checked;
+  expect(readonlySelected).to.equal(
+    expectedLockStatus === LockStatus.ReadonlyAnswers
+  );
+};
+
+describe('LessonLockDialog with stubbed section selector', () => {
+  // Stub out <SectionSelector> so we can test without Redux.
+  let sectionSelectorStub;
+  beforeEach(() => {
+    sectionSelectorStub = sinon.stub(SectionSelector, 'default');
+    sectionSelectorStub.callsFake(props => null);
+  });
+  afterEach(() => {
+    sectionSelectorStub.restore();
+  });
+
+  it('renders with minimal props', () => {
+    const wrapper = mount(<LessonLockDialog {...MINIMUM_PROPS} />);
+    expect(wrapper).not.to.be.null;
+    expect(wrapper.text()).not.to.be.empty;
+  });
 
-describe('LessonLockDialog', () => {
-  it('renders with a selected section', () => {
-    const wrapper = shallow(
-      <LessonLockDialog {...MINIMUM_PROPS} selectedSectionId={fakeSectionId} />
+  it('renders no content when closed', () => {
+    const wrapper = mount(
```

Comment:
Could `shallow` still be used in the cases where `useEffect` doesn't need to be tested?

Reply:
It's possible but I think it's easier when all of the unit tests in a single file (and preferably across a team's components) are all written in a similar way.  I'm also not sure whether `shallow` or `mount` is better.  (See my [response](https://github.com/code-dot-org/code-dot-org/pull/45467/files#r836828312) to Maureen's comment above.)

Reply:
Yeah I guess there's a bigger conversation here. FWIW, in the case where I've been testing `useEffect`, I've been using a mix of `shallow` and `mount` depending on the test, preferring `shallow` when possible. I don't have a strong opinion here though as you're stubbing out the child components

Reply:
I think I would agree if `shallow` and `mount` were more similar but it seems like the style of testing changes quite a bit when changing between them.  This means that a change to a component that affects whether `useEffect` is relevant would require a significant change to unit tests.  I find this to be a pain and negates some of the value of the tests, but I know there are different opinions on how much unit tests can/should depend on the internals of the object under test.  If it's ok with you, I'd like to leave these as all `mount` test for now and continue conversations on `mount` vs. `shallow`.

Reply:
Yep that sounds good to me!